### ISSUE TITLE
Removal of useExperimentalFileWatcher setting regresses file watching behavior

### ISSUE THREAD

**Issue by cbrock** [microsoft/vscode#74630]
- VSCode Version: 1.34.0 (1.34.0)
- OS Version: OSX 10.14.4 (18E226)

After the removal of the `files.useExperimentalFileWatcher` setting (removed as part of https://github.com/microsoft/vscode/pull/71632), changes made to open files by tasks are not reflected in the editor until you leave and come back to the open file.

Previously, using the `files.useExperimentalFileWatcher` setting, changes were immediately reflected in vscode after a task wrote them to disk.

This appears to be a regression and is very frustrating and feels very clumsy (needing to switch away from your current file and back again)

Here is an example of [running a task](https://gist.github.com/cbrock/b9f12b31943ca0b91d634024d1850b6d) which applies Prettier and ESLint to the current file. However the changes are not reflected in the UI until the current file loses, and regains, focus.

![](http://g.recordit.co/3fymiitL3m.gif)

Steps to Reproduce:

1. Create a task that writes to disk. For reference, here is the [task](https://gist.github.com/cbrock/b9f12b31943ca0b91d634024d1850b6d) I use which applies [Prettier](https://prettier.io/) and [ESLint](https://eslint.org/) to the current file
2. Have a .js file with formatting and/or linting errors
3. Run the task
4. Notice that the open file does not reflect the changes written by the task
5. Click away from the current file and back again
6. See the changes reflected in the file.

https://github.com/microsoft/vscode/pull/71632 appears to be the PR that removed the `files.useExperimentalFileWatcher` setting.

Does this issue occur when all extensions are disabled?: Yes

**Comment by alexr00** [comment]
@cbrock It shouldn't be task specific because all tasks does is run your command. I want to confirm that this isn't task specific. Can you try running your task command from the integrated terminal and see if it does the same thing?

**Comment by bpasero** [comment]
@cbrock on top of that question, can you also try to reproduce in our insiders build?  You can give our preview releases a try from: https://code.visualstudio.com/insiders/

**Comment by cbrock** [comment]
> @cbrock on top of that question, can you also try to reproduce in our insiders build? You can give our preview releases a try from: https://code.visualstudio.com/insiders/

@bpasero Per the issue template, I did attempt to reproduce using this insiders build:
```
version: 1.35.0
build: 6b081a9f0089062b2cd105e1f1ab366591291534
```

I tried again this morning using `1.35.0,e9c50663154c369a06355ce752b447af5b580dc3` with the same results.

**Comment by cbrock** [comment]
> @cbrock It shouldn't be task specific because all tasks does is run your command. I want to confirm that this isn't task specific. Can you try running your task command from the integrated terminal and see if it does the same thing?

Thanks for your response @alexr00.

I can confirm that the behavior is the same when running the aforementioned task via the integrated terminal. That is, the open file does not reflect changes until I move focus to a different file and click back again.

**Comment by bpasero** [comment]
@cbrock 
Please see if you can find any logged output that could help us find the source of the issue:
* run `code --verbose` from the command line and attach the output
* in Code select **Help | Toggle Developer Tools** and select the **Console**
* attach any output you see in the **Console** you can find

Thanks!

**Comment by cbrock** [comment]
Thanks for your response @bpasero.

[This gist](https://gist.github.com/cbrock/cc8a6da1bca33d6a27b9cdf3fca55448) contains the logging output from just before I executed the task `fmt`, until the task completed - the same workflow that is illustrated in the gif in my original comment above (and just to be clear, I experienced the original issue again).

Here is a screenshot of the console output as well, just in case it's helpful:

![image](https://user-images.githubusercontent.com/1780846/58818990-0f76ea80-85e4-11e9-9868-e48adb40a4d7.png)

**Comment by bpasero** [comment]
@cbrock these entries are the ones to look at:

![image](https://user-images.githubusercontent.com/900690/58854559-ff234600-869d-11e9-9441-77922eba9563.png)

You can filter the console from the filter box to the top (e.g. type "watcher"). 

To start with: simply open a file, type and save. You should get an entry that the file changed from the watcher. Does that work? And if yes, can you see if you get an entry for when your task has finished to run?

**Comment by cbrock** [comment]
@bpasero Apologies for the delay in responding.

Using your debugging advice, I noticed _a lot_ of File Watcher entries in the logs. Inspecting the entries, I noticed the paths of most of them pointed to a `tmp` directory.

I'm currently working with an [ember-cli](https://ember-cli.com/) project (ember-cli makes heavy use of a project-specific `tmp` directory which holds the app's compiled assets). At the time of this writing, the project's `tmp` directory is 330MB in size.

Somehow, I had not added `tmp` to the `files.watcherExclude` list:

```json
  "files.watcherExclude": {
    "**/tmp/**": true
  }
```

Based on some initial testing, this appears to resolve the issue. My hypothesis is that as the project's `tmp` directory grew in size, the `fmt` task became slower and slower, eventually taking so long as to seem completely unresponsive.

I think this issue can remain closed. Thank you for your help!
